packages/flutter/lib/src/rendering/box.dart

```diff
@@ -1985,7 +1985,7 @@ abstract class RenderBox extends RenderObject {
       }
       return true;
     }());
-    return _size!;
+    return _size ?? (throw AssertionError('RenderBox was not laid out: $this'));
```

Comment:
we should use StateError in release builds, not AssertionError, and we should definitely not call `RenderBox.toString`, because merely referencing that has significant performance implications.

Reply:
1. StateError fixed
2. we should definitely not call RenderBox.toString: May I know a bit more details - do you mean code size or runtime overhead? For code size, I guess toString is not tree-stripped since there can be `dynamic.toString` but I will make an experiment; for runtime overhead, IMHO the whole error processing logic is quite complex and slow, thus no need to worry

Reply:
Hmm I realize in release mode, `toString()` calls `toStringShort` and then it outputs almost nothing (just describeIdentity)...